### PR TITLE
feat: show streaming available

### PR DESCRIPTION
**Pull request by yowpark**
# PR Template

## Checklist

- [ ] E2E finished.
- [x] Local, Self Hosted and Development environment test finished.
- [x] Local, E2E, Self Hosted, Development and Production env variable checked.
- [ ] (Backend only) DB migration file creation checked.
- [ ] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

## Related Issues
- https://github.com/dogu-team/dogu/issues/314
- https://github.com/dogu-team/dogu/issues/297

## Database
- 'display_error' column added on device.entity

## RuntimeInfo
- add 'error' field on runtimeDisplayInfo, and update device's display_error column according to displayInfo.error

### CODE REVIEW

File: packages/typescript/types/src/device/types.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export interface Device {
   isGlobal: number;
   isHost: number;
   isVirtual: number;
+  displayError: string;
```
Comment (Ikalli) [inline packages/typescript/types/src/device/types.ts:41]: i think type for  `string | null` is more accurate for its type

File: projects/console-web-server/src/db/entity/device.entity.ts [hunk 2]
```diff
@@ -71,6 +72,9 @@ export class Device extends BaseEntity implements DeviceBase {
   @Column({ type: 'smallint', name: DevicePropSnake.is_virtual, unsigned: true, default: 0, nullable: false })
   isVirtual!: number;
 
+  @Column({ type: 'character varying', name: DevicePropSnake.display_error, length: DEVICE_DISPLAY_ERROR_MAX_LENGTH, default: '', nullable: false })
+  displayError!: string;
```
Comment (Ikalli) [inline projects/console-web-server/src/db/entity/device.entity.ts:76]: If `displayError` type is `string | null`, `nullable: false` should be removed.

And this action needs to re-generate migration files

File: projects/console-web-server/src/module/private/private-device.controller.ts [hunk 3]
```diff
@@ -183,6 +183,22 @@ export class PrivateDeviceController {
...
+    if (runtimeInfos.length === 0) {
+      return;
+    }
+    const lastInfo = runtimeInfos[runtimeInfos.length - 1];
+    if (0 < lastInfo.displays.length) {
+      const device = await this.deviceRepository.findOne({ where: { deviceId } });
+      if (device) {
+        const errorDisplay = lastInfo.displays.find((display) => display.error.length > 0);
+        if (!errorDisplay && device.displayError.length > 0) {
+          await this.deviceRepository.update({ deviceId }, { displayError: '' });
```
Comment (Ikalli) [inline projects/console-web-server/src/module/private/private-device.controller.ts:195]: If `displayError` type is `string | null`, update it to `null` would be more accurate

**Review by Ikalli — COMMENTED** [review]
What if `displayError` type is `string | null`?

**Comment by oneofthezombies** [comment]
LGTM!

File: projects/console-web-server/src/module/private/private-device.controller.ts (hunk 3 above)
Comment (yowpark) [reply projects/console-web-server/src/module/private/private-device.controller.ts]: @Ikalli I changed `displayError` to nullable! https://github.com/dogu-team/dogu/pull/410/commits/43b53c35920f2e26c11056c5d3224c39c5857c8a

**Pull request merged**